File: branches/BUG-1/src/Servidor/Transaccional/Comandos/Comandos/Receptor.cs

namespace Trascend.Bolet.Comandos.Comandos
{
    public class Receptor<T>
    {
        /// <summary>
        /// Objeto generico a ser almacenado
        /// </summary>
        T _objetoAlmacenado;

        /// <summary>
        /// Propiedad que retorna el objeto genérico almacenado
        /// </summary>
        public T ObjetoAlmacenado
        {
            get { return _objetoAlmacenado; }
        }

        /// <summary>
        /// Constructo predeterminado
        /// </summary>
        /// <param name="objetoRecibido">Objeto a almacenar</param>
        public Receptor(T objetoAlmacenado)
        {
            this._objetoAlmacenado = objetoAlmacenado;
        }
    }
}

File: trunk/src/Servidor/Transaccional/Comandos_Fac/Comandos/Receptor.vb
Namespace Comandos
    Public Class Receptor(Of T)
        ''' <summary>
        ''' Objeto generico a ser almacenado
        ''' </summary>
        Private _objetoAlmacenado As T

        ''' <summary>
        ''' Propiedad que retorna el objeto genérico almacenado
        ''' </summary>
        Public ReadOnly Property ObjetoAlmacenado() As T
            Get
                Return _objetoAlmacenado
            End Get
        End Property

        ' ''' <summary>
        ' ''' Constructo predeterminado
        ' ''' </summary>
        ' ''' <param name="objetoRecibido">Objeto a almacenar</param>
        Public Sub New(ByVal objetoAlmacenado As T)
            Me._objetoAlmacenado = objetoAlmacenado
        End Sub
    End Class
End Namespace